internal/impl/devbox.go

```diff
@@ -348,14 +349,18 @@ func (d *Devbox) Exec(cmds ...string) error {
 	return nix.Exec(nixDir, cmds, env)
 }
 
-func (d *Devbox) PrintShellEnv() error {
-	profileBinDir, err := d.profileBinDir()
+func (d *Devbox) AddPluginEnv() error {
+	pluginEnvs, err := plugin.Env(d.cfg.Packages, d.projectDir)
 	if err != nil {
-		return errors.WithStack(err)
+		return err
+	}
+	script := ""
+	for _, pluginEnv := range pluginEnvs {
+		script += fmt.Sprintf("export %s\n", pluginEnv)
 	}
-	// TODO: For now we just updated the PATH but this may need to evolve
-	// to essentially a parsed shellrc.tmpl
-	fmt.Fprintf(d.writer, "export PATH=\"%s:$PATH\"", profileBinDir)
+	// we have to use stdout here because d.writer is set to output to stderr
+	// which makes direnv unable to read the output
+	fmt.Fprintln(os.Stdout, script)
```

Comment:
Instead of printing to stdout in a library, what if this returns the script? and in `runShellCmd` you can do `cmd.Print()` ?

Reply:
If we do this we should rename this function, maybe just `PluginEnv`

Reply:
Good suggestion. I'll make the change and push it now.